internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -650,7 +649,7 @@ func setResourceRootFields(d *schema.ResourceData, cluster *admin.ClusterDescrip
 		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "version_release_system", clusterName, err))
 	}
 
-	if err := d.Set("accept_data_risks_and_force_replica_set_reconfig", conversion.TimePtrToStringPtr(cluster.AcceptDataRisksAndForceReplicaSetReconfig)); err != nil {
+	if err := d.Set("accept_data_risks_and_force_replica_set_reconfig", conversion.TimePtrToStringPtr(cluster.AcceptDataRisksAndForceReplicaSetReconfig)); isResourceSchema && err != nil {
```

Comment:
should isResourceSchema be before d.Set()? 
Or we are simply ignoring the error from d.Set when isResourceSchema is set?

Reply:
ya currently it just ignores the error, no strong opinion, I'll update it

Reply:
I do see that `accept_data_risks_and_force_replica_set_reconfig` and `cluster_id` are not part of the data source schema, not sure if there is a risk in setting the values if not defined in the schema.

Reply:
updated